src/codec.rs

```diff
@@ -31,16 +31,23 @@ pub mod pb_codec {
 
     use super::MessageReader;
     use crate::buf::GrpcSlice;
-    use crate::error::Result;
+    use crate::error::{Error, Result};
 
     #[inline]
-    pub fn ser<T: Message>(t: &T, buf: &mut GrpcSlice) {
-        let cap = t.compute_size();
-        unsafe {
-            let bytes = buf.realloc(cap as usize);
-            let raw_bytes = &mut *(bytes as *mut [std::mem::MaybeUninit<u8>] as *mut [u8]);
-            let mut s = CodedOutputStream::bytes(raw_bytes);
-            t.write_to_with_cached_sizes(&mut s).unwrap();
+    pub fn ser<T: Message>(t: &T, buf: &mut GrpcSlice) -> Result<()> {
+        let cap = t.compute_size() as usize;
+        // FIXME: This is not a practical fix until stepancheg/rust-protobuf#530 is fixed.
+        if cap <= u32::MAX as usize {
+            unsafe {
+                let bytes = buf.realloc(cap);
+                let raw_bytes = &mut *(bytes as *mut [std::mem::MaybeUninit<u8>] as *mut [u8]);
+                let mut s = CodedOutputStream::bytes(raw_bytes);
+                t.write_to_with_cached_sizes(&mut s).map_err(Into::into)
+            }
+        } else {
+            Err(Error::Codec(
+                format!("message is too large: {} > u32::MAX", cap).into(),
```

Comment:
Add comment to explain the limit is from somewhere in grpc core

Reply:
refer to https://github.com/tikv/grpc-rs/pull/496/files#diff-ca14eabd5ab83f72548cd7e44eaea3f56a5d2808e3275c6b2a97622ebe5d1dd0R9-R11 ?